iOSClient/Main/Collection Common/NCCollectionViewCommon.swift

```diff
@@ -1668,12 +1659,11 @@ extension NCCollectionViewCommon: UICollectionViewDataSource {
         }
 
         // ** IMPORT MUST BE AT THE END **
-        //
-
-        if !metadata.isSharable() {
+        // Disable Share Button
+        if appDelegate.disableSharesView || !metadata.isSharable() {
             cell.hideButtonShare(true)
         }
-        
+
         return cell
```

Comment:
If this should always be at the end, can't we put it in a `defer` block and move it to the top of this function?